components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/CommonUtil.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.http.conn.params.ConnRoutePNames;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.woden.WSDLFactory;
+import org.apache.woden.WSDLReader;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import javax.wsdl.WSDLException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class CommonUtil {
+    private static final Log log = LogFactory.getLog(CommonUtil.class);
+
+    private static void handleException(String msg) throws APIManagementException {
+        log.error(msg);
+        throw new APIManagementException(msg);
+    }
+
+    private static void handleException(String msg, Throwable t) throws APIManagementException {
+        log.error(msg, t);
+        throw new APIManagementException(msg, t);
+    }
+
+    public static void validateWsdl(String url) throws APIManagementException, IOException {
+        BufferedReader in = null;
+        try {
+            URL wsdl = new URL(url);
+            in = new BufferedReader(new InputStreamReader(wsdl.openStream()));
+
+            String inputLine;
+            boolean isWsdl2 = false;
+            boolean isWsdl10 = false;
+            StringBuilder urlContent = new StringBuilder();
+            while ((inputLine = in.readLine()) != null) {
+                String wsdl2NameSpace = "http://www.w3.org/ns/wsdl";
+                String wsdl10NameSpace = "http://schemas.xmlsoap.org/wsdl/";
+                urlContent.append(inputLine);
+                isWsdl2 = urlContent.indexOf(wsdl2NameSpace) > 0;
+                isWsdl10 = urlContent.indexOf(wsdl10NameSpace) > 0;
+            }
+            if (isWsdl10) {
+                javax.wsdl.xml.WSDLReader wsdlReader11 = null;
+                try {
+                    wsdlReader11 = javax.wsdl.factory.WSDLFactory.newInstance().newWSDLReader();
+                    wsdlReader11.readWSDL(url);
+                } catch (WSDLException e) {
+                    handleException("Unable to process wsdl 1.0", e);
+                }
+
+            } else if (isWsdl2) {
+                WSDLReader wsdlReader20 = null;
+                try {
+                    wsdlReader20 = WSDLFactory.newInstance().newWSDLReader();
+                    wsdlReader20.readWSDL(url);
+                } catch (org.apache.woden.WSDLException e) {
+                    handleException("Unable to process wsdl 2.0", e);
+                }
+
+            } else {
+                handleException("URL is not in format of wsdl1/wsdl2");
+            }
+        } finally {
+            in.close();
+        }
+    }
+
+    /**
+     * Validate the backend by sending HTTP HEAD
+     *
+     * @param urlVal - backend URL
+     * @return - status of HTTP HEAD Request to backend
+     */
+    public static String sendHttpHEADRequest(String urlVal) {
```

Comment:
Not sure why returning String to check for connectivity. Any specific reason? Isn't returning a boolean work here?